src/mlpack/methods/ann/layer/bilinear_interpolation_impl.hpp

```diff
@@ -22,168 +23,198 @@ namespace ann /** Artificial Neural Network. */ {
+        }
       }
     }
   }
-}
 
 template<typename InputDataType, typename OutputDataType>
 template<typename eT>
 void BilinearInterpolation<InputDataType, OutputDataType>::Backward(
-    const arma::Mat<eT>& /*input*/,
-    const arma::Mat<eT>& gradient,
-    arma::Mat<eT>& output)
-{
-  if (output.is_empty())
-    output.set_size(inRowSize * inColSize * depth, batchSize);
-  else
+  const arma::Mat<eT>& /*input*/,
+  const arma::Mat<eT>& gradient,
+  arma::Mat<eT>& output)
   {
-    assert(output.n_rows == inRowSize * inColSize * depth);
-    assert(output.n_cols == batchSize);
-  }
+    if (output.is_empty())
+      output.set_size(inRowSize * inColSize * depth, batchSize);
+    else
+    {
+      assert(output.n_rows == inRowSize * inColSize * depth);
+      assert(output.n_cols == batchSize);
+    }
 
-  assert(outRowSize >= 2);
-  assert(outColSize >= 2);
+    assert(outRowSize >= 2);
+    assert(outColSize >= 2);
 
-  arma::cube gradientAsCube(((arma::Mat<eT>&) gradient).memptr(), outRowSize,
+    arma::cube gradientAsCube(((arma::Mat<eT>&) gradient).memptr(), outRowSize,
       outColSize, depth * batchSize, false, false);
-  arma::cube outputAsCube(output.memptr(), inRowSize, inColSize,
-                          depth * batchSize, false, true);
-
-  if (gradient.n_elem == output.n_elem)
-  {
-    outputAsCube = gradientAsCube;
-  }
-  else
-  {
-    double scaleRow = (double)(outRowSize) / inRowSize;
-    double scaleCol = (double)(outColSize) / inColSize;
-
-    arma::mat22 coeffs;
-    for (size_t i = 0; i < inRowSize; ++i)
+    arma::cube outputAsCube(output.memptr(), inRowSize, inColSize,
+      depth * batchSize, false, true);
+    // Padded Output matrix
+    arma::mat temp(inRowSize + 2, inColSize + 2);
+    if (gradient.n_elem == output.n_elem)
+    {
+      outputAsCube = gradientAsCube;
+    }
+    else
     {
-      size_t rOrigin = (size_t) std::floor(i * scaleRow);
-      if (rOrigin > outRowSize - 2)
-        rOrigin = outRowSize - 2;
-      double deltaR = i * scaleRow - rOrigin;
-      for (size_t j = 0; j < inColSize; ++j)
+      for (size_t k = 0; k < depth * batchSize; ++k)
       {
-        size_t cOrigin = (size_t) std::floor(j * scaleCol);
-
-        if (cOrigin > outColSize - 2)
-          cOrigin = outColSize - 2;
-
-        double deltaC = j * scaleCol - cOrigin;
-        coeffs[0] = (1 - deltaR) * (1 - deltaC);
-        coeffs[1] = deltaR * (1 - deltaC);
-        coeffs[2] = (1 - deltaR) * deltaC;
-        coeffs[3] = deltaR * deltaC;
-
-        for (size_t k = 0; k < depth * batchSize; ++k)
+        temp.zeros();
+        for (size_t i = 0; i < outRowSize; ++i)
         {
-          outputAsCube(i, j, k) = arma::accu(gradientAsCube.slice(k).submat(
-              rOrigin, cOrigin, rOrigin + 1, cOrigin + 1) % coeffs);
+          for (size_t j = 0; j < outColSize; ++j)
+          {
+            size_t current_idx = j * outRowSize + i;
+            // Bottom right corner of the mapped unit matrix
```

Comment:
```suggestion
            // Bottom right corner of the mapped unit matrix.
```

Pedantic style issue, missing stop at the end.